### PR TITLE
Restore Windows CI runs

### PR DESCRIPTION
**Pull request by basil**
Downstream of #6026, #6029, #6030, #6034, #6035, and jenkinsci/jenkins-test-harness#357. With all those changes and with the topmost commit of this PR, the tests [pass on Windows in CI](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6024/25/).

### CODE REVIEW

**Comment by MarkEWaite** [comment]
Likely will hit test timeouts because the tests are detectably slower on Windows than on Linux

**Comment by basil** [comment]
Well with the changes from #6026 I got the smoke tests passing locally, but I can't seem to get past the unit tests in CI. The unit tests seem to hang at the end before the build proceeds to building the WAR and running the integration tests. Since I can't reproduce this locally any help would be appreciated.

**Comment by basil** [comment]
> Well with the changes from #6026 I got the smoke tests passing locally, but I can't seem to get past the unit tests in CI. The unit tests seem to hang at the end before the build proceeds to building the WAR and running the integration tests. Since I can't reproduce this locally any help would be appreciated.

Interestingly enough the hang on CI was in `LoadStatisticsTest#graph`, which passes for me locally on Windows. Not sure why it fails on CI, but for now I can get the CI build to move forward by skipping that test.

**Comment by timja** [comment]
1hr 20 slower than linux 😢

**Comment by basil** [comment]
Caught a bug in #6025 before it ever shipped, so worth it I suppose

**Comment by jglick** [comment]
https://github.com/jenkinsci/jenkins/pull/4447#issuecomment-589172726 FTR

File: Jenkinsfile
Comment (jglick) [inline Jenkinsfile:33]: Do we really need four combinations? To save resources, and sometimes wall clock time, I think it would suffice to run, say, Linux builds on Java 11 and Windows builds on Java 8.

Comment (basil) [reply Jenkinsfile:33]: Nah, I was planning to reduce this to two runs before marking this change as ready for review. But in the meantime I've been running the draft PR with 4 builds to shake out any flakiness.

BTW jenkinsci/jenkins#6029

Comment (basil) [reply Jenkinsfile:33]: My bad. I marked this as draft.

**Review by jglick — APPROVED** [review]
:tada:

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
